src/neo-vm/Types/Struct.cs

```diff
@@ -52,13 +51,13 @@ public override StackItem ConvertTo(StackItemType type)
             return base.ConvertTo(type);
         }
 
-        public override bool Equals(object obj)
+        public override bool Equals(StackItem other)
         {
-            if (!(obj is Struct other)) return false;
+            if (!(other is Struct s)) return false;
```

Comment:
Should we add reference equality checking? If it points to the same object in memory then we can skip iterating over all internal items.

Reply:
There is a reference equality checking in L67.

Reply:
ah yes. We can ignore this comment :)